src/refdb_fs.c

```diff
@@ -1170,20 +1170,18 @@ static int cmp_old_ref(int *cmp, git_refdb_backend *backend, const char *name,
  * check with HEAD only which should cover 99% of all usage
  * scenarios (even 100% of the default ones).
  */
-static int maybe_append_head(refdb_fs_backend *backend, const git_reference *ref, const git_signature *who, const char *message)
+static int maybe_append_head(
+	refdb_fs_backend *backend, const git_reference *ref,
+	const git_oid *old_id, const git_oid *new_id,
+	const git_signature *who, const char *message)
 {
 	int error;
-	git_oid old_id;
 	git_reference *tmp = NULL, *head = NULL, *peeled = NULL;
 	const char *name;
 
 	if (ref->type == GIT_REF_SYMBOLIC)
 		return 0;
 
-	/* if we can't resolve, we use {0}*40 as old id */
-	if (git_reference_name_to_id(&old_id, backend->repo, ref->name) < 0)
-		memset(&old_id, 0, sizeof(old_id));
```

Comment:
Previously, we've always passed a valid old ID into `reflog_append`, where we now always pass `NULL` to it (as there seems to be only one caller, which passes `NULL`). This is no problem, though, as `reflog_append` does the exact same dance that is getting deleted here in case `old` is `NULL`.

Reply:
Not sure what to make of that, so thanks, I guess 😉? My rational argument would be that it moves the duties of formatting NULL to `reflog_append`.

Reply:
I'm always in favor of more tests :)